src/Microsoft.IdentityModel.Tokens/Encryption/SymmetricKeyWrapProvider.cs

```diff
@@ -66,13 +66,13 @@ public SymmetricKeyWrapProvider(SecurityKey key, string algorithm)
             if (string.IsNullOrEmpty(algorithm))
                 throw LogHelper.LogArgumentNullException(nameof(algorithm));
 
-            if (!IsSupportedAlgorithm(key, algorithm))
+            if (!SupportedAlgorithms.IsSupportedSymmetricKeyWrap(algorithm, key))
                 throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10661, algorithm, key)));
 
             Algorithm = algorithm;
             Key = key;
 
-            _symmetricAlgorithm = GetSymmetricAlgorithm(key, algorithm);
+            _symmetricAlgorithm = GetSymetricAlgorithmInternal(key, algorithm);
```

Comment:
I agree that this represents bad design and a code smell, but we might break some users who are leveraging this approach. Wondering if we could instead resolve an algorithm once (before use) and not invoke the method from the constructor? This seems to be harder to accomplish with AsymmetricSignatureProvider, though.